azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/database/JdbcUrl.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.database;
+
+import com.google.common.base.Preconditions;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.Objects;
+
+public class JdbcUrl {
+
+    private static final int MYSQL_DEFAULT_PORT = 3306;
+    private static final int SQLSERVER_DEFAULT_PORT = 1433;
+
+    private final URIBuilder uri;
+    private String username;
+    private String password;
+
+    private JdbcUrl(String url) {
+        Preconditions.checkArgument(StringUtils.startsWith(url, "jdbc:"), "invalid jdbc url.");
+        String convertedUrl = url;
+        if (StringUtils.startsWith(url, "jdbc:sqlserver:")) {
+            convertedUrl = StringUtils.replaceOnce(url, ";", "?").replaceAll(";", "&");
+        }
+        try {
+            this.uri = new URIBuilder(convertedUrl.substring(5));
+        } catch (final URISyntaxException e) {
+            throw new AzureToolkitRuntimeException("invalid jdbc url: %s", url);
+        }
+    }
+
+    public static JdbcUrl from(String connectionString) {
+        Preconditions.checkArgument(StringUtils.startsWith(connectionString, "jdbc:"), "invalid jdbc url.");
```

Comment:
May remove the pre check as it will be checked in constructor

Reply:
updated.